packages/text-bitmap/src/BitmapFontFactory.ts

```diff
@@ -0,0 +1,221 @@
+            };
+            fontData.page[id] = {
+                id,
+                file: '',
+            };
+
+            positionX += (width + (2 * BMT_PADDING)) * resolution;
+        }
+
+        return new BitmapFont(fontData, textures);
+    }
+
+    private static render(
+        canvas: HTMLCanvasElement,
+        context: CanvasRenderingContext2D,
+        metrics: TextMetrics,
+        x: number,
+        y: number,
+        resolution: number,
+        style: TextStyle
+    ): void
+    {
+        const char = metrics.text;
+        const fontProperties = metrics.fontProperties;
+
+        context.translate(x, y);
+        context.scale(resolution, resolution);
+
+        const tx = style.strokeThickness / 2;
+        const ty = -(style.strokeThickness / 2);
+
+        context.font = style.toFontString();
+        context.lineWidth = style.strokeThickness;
+        context.textBaseline = style.textBaseline;
+        context.lineJoin = style.lineJoin;
+        context.miterLimit = style.miterLimit;
+
+        // set canvas text styles
+        context.fillStyle = generateFillStyle(canvas, context, style, resolution, [char], metrics);
+        context.strokeStyle = style.stroke as string;
+
+        context.font = style.toFontString();
+        context.lineWidth = style.strokeThickness;
+        context.textBaseline = style.textBaseline;
+        context.lineJoin = style.lineJoin;
+        context.miterLimit = style.miterLimit;
+
+        // set canvas text styles
+        context.fillStyle = generateFillStyle(canvas, context, style, resolution, [char], metrics);
+        context.strokeStyle = style.stroke as string;
+
+        const dropShadowColor = style.dropShadowColor;
+        const rgb = hex2rgb(typeof dropShadowColor === 'number' ? dropShadowColor : string2hex(dropShadowColor));
+
+        if (style.dropShadow)
+        {
+            context.shadowColor = `rgba(${rgb[0] * 255},${rgb[1] * 255},${rgb[2] * 255},${style.dropShadowAlpha})`;
+            context.shadowBlur = style.dropShadowBlur;
+            context.shadowOffsetX = Math.cos(style.dropShadowAngle) * style.dropShadowDistance;
+            context.shadowOffsetY = (Math.sin(style.dropShadowAngle) * style.dropShadowDistance);
+        }
+        else
+        {
+            context.shadowColor = '0';
+            context.shadowBlur = 0;
+            context.shadowOffsetX = 0;
+            context.shadowOffsetY = 0;
+        }
+
+        if (style.stroke && style.strokeThickness)
+        {
+            context.strokeText(char, tx, ty + metrics.lineHeight - fontProperties.descent);
+        }
+        if (style.fill)
+        {
+            context.fillText(char, tx, ty + metrics.lineHeight - fontProperties.descent);
+        }
+
+        context.setTransform();
+
+        context.fillStyle = 'rgba(0, 0, 0, 0)';
+    }
+}
+
+export interface IBitmapFontFactoryOptions extends ITextStyle
+{
+    chars: string | string[];
+    resolution: number;
+}
```

Comment:
I would change these options to require `chars`.
```suggestion
export interface IBitmapFontFactoryOptions extends Partial<ITextStyle>
{
    chars: string | string[];
    resolution?: number;
}
```